make/buf/scripts/release.bash

```diff
@@ -11,137 +11,42 @@ fail() {
   exit 1
 }
 
-goos() {
-  case "${1}" in
-    Darwin) echo darwin ;;
-    Linux) echo linux ;;
-    Windows) echo windows ;;
-    *) return 1 ;;
-  esac
-}
-
-goarch() {
-  case "${1}" in
-    x86_64) echo amd64 ;;
-    arm64) echo arm64 ;;
-    aarch64) echo arm64 ;;
-    *) return 1 ;;
-  esac
-}
+[ -n "${RELEASE_GO_VERSION}" ] || fail "RELEASE_GO_VERSION must be set."
 
-sha256() {
-  if ! type sha256sum >/dev/null 2>/dev/null; then
-    if ! type shasum >/dev/null 2>/dev/null; then
-      echo "sha256sum and shasum are not installed" >&2
-      return 1
-    else
-      shasum -a 256 "$@"
-    fi
-  else
-    sha256sum "$@"
-  fi
-}
-
-if [ -z "${INSIDE_DOCKER}" ]; then
-  if [ -z "${RELEASE_MINISIGN_PRIVATE_KEY}" -o -z "${RELEASE_MINISIGN_PRIVATE_KEY_PASSWORD}" ]; then
-    fail "RELEASE_MINISIGN_PRIVATE_KEY and RELEASE_MINISIGN_PRIVATE_KEY_PASSWORD must be set."
-  fi
-  if [ -z "${DOCKER_IMAGE}" ]; then
-    fail "DOCKER_IMAGE must be set"
-  fi
-  docker run --volume \
-    "${DIR}:/app" \
-    --workdir "/app" \
-    --rm \
-    -e INSIDE_DOCKER=1 \
-    "${DOCKER_IMAGE}" \
-    bash -x make/buf/scripts/release.bash
-  if [ "$(uname -s)" == "Linux" ]; then
-    sudo chown -R "$(id -u):$(id -g)" .build
-  fi
-  # Produce the signature outside the docker image where we have
-  # minisign installed.
-  secret_key_file="$(mktemp)"
-  trap "rm ${secret_key_file}" EXIT
-  # Prevent printing of private key and password
-  set +x
-  echo "${RELEASE_MINISIGN_PRIVATE_KEY}" > "${secret_key_file}"
-  echo "${RELEASE_MINISIGN_PRIVATE_KEY_PASSWORD}" | minisign -S -s "${secret_key_file}" -m .build/release/buf/assets/sha256.txt
-  set -x
-  exit 0
+if [ -n "${RELEASE_SKIP_SIGN}" ]; then
+  SKIP_SIGN="--skip-sign"
+elif [ -z "${RELEASE_MINISIGN_PRIVATE_KEY}" -o -z "${RELEASE_MINISIGN_PRIVATE_KEY_PASSWORD}" ]; then
+  fail "RELEASE_MINISIGN_PRIVATE_KEY and RELEASE_MINISIGN_PRIVATE_KEY_PASSWORD must be set."
 fi
 
-BASE_NAME="buf"
+[ -z "${RELEASE_SKIP_VALIDATE}" ] || SKIP_VALIDATE="--skip-validate"
+
+RELEASE_DIR=".build/release/buf"
+ASSETS_DIR="${RELEASE_DIR}/assets"
+WORKSPACE_DIR="${RELEASE_DIR}/workspace"
+DIST_DIR="${WORKSPACE_DIR}/dist"
 
-RELEASE_DIR=".build/release/${BASE_NAME}"
 rm -rf "${RELEASE_DIR}"
-mkdir -p "${RELEASE_DIR}"
-cd "${RELEASE_DIR}"
+mkdir -p "${ASSETS_DIR}" "${WORKSPACE_DIR}"
+trap "rm -rf $WORKSPACE_DIR" EXIT
+
+test -f "${GOBIN}/${RELEASE_GO_VERSION}" || go install "golang.org/dl/${RELEASE_GO_VERSION}"@latest
+"${GOBIN}/${RELEASE_GO_VERSION}" download 2>/dev/null
```

Comment:
Usage of `GOBIN` in this script scares me, this is the project `$GOBIN` right? The release process should be completely isolated, similar to how it was with Docker